From: Dimitri John Ledkov <xnox@ubuntu.com>
Date: Wed, 25 May 2022 18:22:01 +0800
Subject: Handle late installation of systemd/resolvconf gracefully, by not

reverting /etc/resolv.conf symlink to a backup of the backup.
---
 scripts/build/lb_chroot_resolv | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/scripts/build/lb_chroot_resolv b/scripts/build/lb_chroot_resolv
index 9a7a052..438bae0 100755
--- a/scripts/build/lb_chroot_resolv
+++ b/scripts/build/lb_chroot_resolv
@@ -81,6 +81,15 @@ case "${1}" in
 			# Copying local resolv.conf
 			cp -a config/includes.chroot/etc/resolv.conf chroot/etc/resolv.conf
 			rm -f chroot/etc/resolv.conf.orig
+		elif [ -L chroot/etc/resolv.conf ] && [ -e chroot/etc/resolv.conf.orig ]
+		then
+			# After debootstrap (which copied resolv.conf from
+			# the host) and after we backed that up, something
+			# converted /etc/resolv.conf into a
+			# symlink. E.g. a late installation of
+			# systemd-resolved or resolvconf. Do not touch the
+			# new symlink, but do clean up our backup.
+			rm -f chroot/etc/resolv.conf.orig
 		elif [ -e chroot/etc/resolv.conf.orig ] || [ -L chroot/etc/resolv.conf.orig ]
 		then
 			# Restoring resolv file or symlink
